widgetastic_manageiq.py

```diff
@@ -1108,3 +1111,104 @@ def select(self, title):
     @property
     def selected(self):
         return next(btn.title for btn in self._view_buttons if btn.active)
+
+
+class TimelinesFilter(View):
+    """represents Filter Part of Timelines view
+
+    """
+    # common
+    event_type = BootstrapSelect(id='tl_show')
+    event_category = BootstrapSelect(id='tl_category_management')
+    time_period = Stepper(locator='//div[contains(@class, "timeline-stepper")]')
+    time_range = BootstrapSelect(id='tl_range')
+    time_position = BootstrapSelect(id='tl_timepivot')
+    # date_picker = Calendar()
+    apply = Text(locator='.//div[contains(@class, "timeline-apply")]')
+    # management controls
+    detailed_events = Checkbox(name='showDetailedEvents')
+    # policy controls
+    policy_event_category = BootstrapSelect(id='tl_category_policy')
+    policy_event_status = RadioGroup(locator='//span[contains(@class, "timeline-option")]')
+
+
+class TimelinesChart(View):
+    """represents Chart part of Timelines View
+
+    # currently only event collection is available
+    # todo: to add widgets for all controls
+    """
+    ROOT = '//div[contains(@class, "timeline-container")]'
+    CATEGORIES = './/*[name()="g" and contains(@class, "timeline-pf-labels")]' \
+                 '//*[name()="text" and @class="timeline-pf-label"]'
+
+    EVENTS = '(.//*[name()="g" and contains(@class, "timeline-pf-drops-container")]/*[name()="g" ' \
+             'and @class="timeline-pf-drop-line"])[{pos}]/*[name()="text" ' \
+             'and @class="timeline-pf-drop"]'
+    # LEGEND = ''
+    # plus_zoom = Button('+')
+    # minus_zoom = Button('-')
+
+    def __init__(self, parent, logger=None):
+        super(TimelinesChart, self).__init__(parent=parent, logger=logger)
+
+    def get_categories(self, *categories):
+        br = self.browser
+        prepared_categories = []
+        for num, element in enumerate(br.elements(self.CATEGORIES), start=1):
+            # categories have number of events inside them
+            mo = re.search('^(.*?)(\s\(\s*\d+\s*\)\s*)*$', element.text)
```

Comment:
Use self.browser.text(element) instead of element.text. In ideal case it has no extra selenium calls. It only makes extra selenium calls when selenium is grumpy and does not want to return the text (when an element is obscured)